fix/3795: Stop the link modal from triggering maybeStartTyping

BoardJames opened the pull request:
## Description
When the toolbar is attached to the block any typing in the link modal would trigger the maybeStartTyping event which would hide the toolbar and in process also hide the link modal. This fixes the problem by suppressing the KeyPress event.

Fixes #3795 

## How Has This Been Tested?
Manually tested in Firefox and Chrome on Linux mint.

## Types of changes
Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

tiny-james committed:
fix/3795: Stop the link modal from triggering maybeStartTyping

When the toolbar is attached to the block any typing in the link modal
would trigger the maybeStartTyping event which would hide the toolbar
and in process also hide the link modal.

tiny-james committed:
fix/3795: Renable eslint rules

noisysocks commented on blocks/editable/format-toolbar/index.js:
```diff
@@ -42,6 +42,13 @@ const FORMATTING_CONTROLS = [
+	 a maybeStartTyping on keypress, and that hides the "fixed-to-block" toolbar
+	 which unregisters the slot, so when Editable tries to re-render its input
+	 dialog, the slot is no longer in the system, and the dialog disappears
+ */
```
nit: Could you shorten this to a one line comment? It's enough that the comment gives the reader a quick heads up so that they can e.g. run a `git blame` if they'd like to learn more about the bug fix.

My suggestion:

```js
// Stop the keypress event from propagating up to maybeStartTyping in BlockListBlock
const stopKeyPressPropagation = ( event ) => event.stopPropagation();
```

noisysocks reviewed (approved):
Thanks for looking into this! 

👍 I was able to reproduce the issue and confirm that this PR fixes it.

tiny-james committed:
fix/3795: Simplified comment

== Original comment ==
We need to stop the keypress event here [in format toolbar], because
block.js is firing a maybeStartTyping on keypress, and that hides the
"fixed-to-block" toolbar which unregisters the slot, so when Editable
tries to re-render its input dialog, the slot is no longer in the
system, and the dialog disappears
